gr-digital/include/gnuradio/digital/ofdm_cyclic_prefixer.h

```diff
@@ -45,28 +45,37 @@ namespace gr {
      *         the pulse shaping.
      *
      * The pulse shape is a raised cosine in the time domain.
+     *
+     * Different CP lengths as for instance needed in LTE are supported. This
+     * is why one of the inputs is std::vector<int>. After every CP given has
+     * been prepended to symbols, each with the length of the IFFT operation,
+     * the mechanism will wrap arround and start over. To give an example, the
+     * input tuple for LTE with an FFT length of 2048 would be (160,) +
+     * (144,)*6, which is equal to (160, 144, 144, 144, 144, 144, 144). A
+     * uniform CP would be indicated by (uniform_cp_length, ).
      */
+
     class DIGITAL_API ofdm_cyclic_prefixer : virtual public tagged_stream_block
     {
      public:
       typedef boost::shared_ptr<ofdm_cyclic_prefixer> sptr;
 
       /*!
-       * \param input_size FFT length (i.e. length of the OFDM symbols)
-       * \param output_size FFT length + cyclic prefix length (in samples)
-       * \param rolloff_len Length of the rolloff flank in samples
-       * \param len_tag_key For framed processing the key of the length tag
+       * \param fft_len IFFT length (i.e. length of the OFDM symbols).
+       * \param cp_lengths CP lengths. Wraps arround after reaching the end.
+       * \param rolloff_len Length of the rolloff flank in samples.
+       * \param len_tag_key For framed processing the key of the length tag.
        */
-      static sptr make(
-	  size_t input_size,
-	  size_t output_size,
-	  int rolloff_len=0,
-	  const std::string &len_tag_key=""
+
+      static sptr make
+      (
+        int fft_len,
+        std::vector<int> cp_lengths,
```

Comment:
This is a pretty massive API change. I think we can make it a much less problematic change if you add another, overloaded `make()` call with the old signature. We have that elsewhere.